### PR TITLE
Drop support for k3s 1.19 or lower, bump calico to 3.21 for k3s 1.20, and calico 3.23 for k3s 1.21+

### PR DESCRIPTION
**Pull request by consideRatio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 [![GitHub Action badge](https://github.com/jupyterhub/action-k3s-helm/workflows/Test/badge.svg)](https://github.com/jupyterhub/action-k3s-helm/actions)
 
 Creates a Kubernetes cluster using [K3s](https://k3s.io/) (1.17+) with
-[Calico](https://www.projectcalico.org/) (3.20) for
+[Calico](https://www.projectcalico.org/) (3.22) for
```
Comment (consideRatio) [inline README.md:6]:
```suggestion
[Calico](https://www.projectcalico.org/) (3.23) for
```

File: action.yml [hunk 2]
```diff
@@ -134,10 +134,10 @@ runs:
     #
     # ref: https://rancher.com/docs/k3s/latest/en/installation/network-options/
     #
-    - name: Setup calico v3.20
+    - name: Setup calico v3.22
       run: |
-        echo "::group::Setup calico v3.20"
-        curl -sfL --output /tmp/calico.yaml https://docs.projectcalico.org/v3.20/manifests/calico.yaml
+        echo "::group::Setup calico v3.22"
+        curl -sfL --output /tmp/calico.yaml https://docs.projectcalico.org/v3.22/manifests/calico.yaml
```
Comment (consideRatio) [inline action.yml:140]:
```suggestion
        echo "::group::Setup calico v3.23"
        curl -sfL --output /tmp/calico.yaml https://docs.projectcalico.org/v3.23/manifests/calico.yaml
```

**Comment by manics** [comment]
From https://projectcalico.docs.tigera.io/getting-started/kubernetes/requirements

> Kubernetes requirements
> Supported versions
> 
> We test Calico v3.23 against the following Kubernetes versions.
> 
>     v1.21
>     v1.22
>     v1.23
> 
> Due to changes in the Kubernetes API, Calico v3.23 will not work on Kubernetes v1.15 or below. v1.16-v1.18 may work, but they are no longer tested.

**Comment by manics** [comment]
Sounds good to me! If it's a major bump it's probably worth going reviewing all components and bumping them if they're EOL

**Comment by consideRatio** [comment]
@manics I'm pushing this onward to a v3 release now.

- I had to use an older version for calico in k3s 1.20 because a poddisruptionbudget resource was declared with too modern api (policy/v1, instead of policy/v1beta1).
- I had to use calico v3.21 instead of v3.22 because it isn't available any more due to a bug of their to redirect to some new domain etc. I have not yet reported it, yet.

**Pull request merged**